rocAL/include/loaders/loader_module.h

```diff
@@ -53,8 +53,8 @@ class LoaderModule {
     virtual Timing timing() = 0;                    // Returns timing info
     virtual std::vector<std::string> get_id() = 0;  // returns the id of the last batch of images/frames loaded
     virtual void start_loading() = 0;               // starts internal loading thread
-    virtual decoded_image_info get_decode_image_info() = 0;
-    virtual crop_image_info get_crop_image_info() = 0;
+    virtual decoded_sample_info get_decode_sample_info() = 0;
+    virtual crop_image_info get_crop_image_info() { return {}; }
```

Comment:
Why changed to virtual?

If it needs to be done, then set random bbox can also be made virtual

Also check if we need to throw error for virtual functions

Reply:
Its changed to virtual because we do not need the crop_image_info in audio cases and cifar loader cases

Reply:
For 

void set_random_bbox_data_reader(std::shared_ptr<RandomBBoxCrop_MetaDataReader> randombboxcrop_meta_data_reader) override { THROW("set_random_bbox_data_reader is not compatible with this implementation") };

It is used in the audio_loader.h files. For other cases, handled already. 

This PR is specific for the name changes wrt circulatr buffer variable


Can we handle the random bbox reader case seperately as a part of a diff PR?

@shobana-mcw and @fiona-gladwin please comment